js/widgets/temperament.js

```diff
@@ -2169,11 +2157,68 @@ class TemperamentWidget {
                 i -= 1;
             }
 
-            if (i <= pitchNumber && i >= 0 && this._playing && p < 2) {
+            if (i <= pitchNumber && i >= 0 && p < 2) {
                 setTimeout(() => {
-                    __playLoop(i);
+                    if (this.closed) {
+                        this.closed = !this.closed;
+                        return;
+                    } else if (!this._playing) {
+                        cell.innerHTML =
+                            '&nbsp;&nbsp;<img src="header-icons/' +
+                            "play-button.svg" +
+                            '" title="' +
+                            _("Play") +
+                            '" alt="' +
+                            _("Play") +
+                            '" height="' +
+                            TemperamentWidget.ICONSIZE +
+                            '" width="' +
+                            TemperamentWidget.ICONSIZE +
+                            '" vertical-align="middle" align-content="center">&nbsp;&nbsp;';
+                        if (i !== -1) {
+                            if (this.circleIsVisible == false && docById("wheelDiv4") == null) {
+                                this.notesCircle.navItems[i - 1].fillAttr = "#c8C8C8";
+                                this.notesCircle.navItems[i - 1].sliceHoverAttr.fill = "#c8C8C8";
+                                this.notesCircle.navItems[i - 1].slicePathAttr.fill = "#c8C8C8";
+                                this.notesCircle.navItems[i - 1].sliceSelectedAttr.fill = "#c8C8C8";
+                                if (i == 11) {
+                                    //on completion of a full circle and on hitting '0' note in clockwise direction
+                                    this.notesCircle.navItems[0].fillAttr = "#c8C8C8";
+                                    this.notesCircle.navItems[0].sliceHoverAttr.fill = "#c8C8C8";
+                                    this.notesCircle.navItems[0].slicePathAttr.fill = "#c8C8C8";
+                                    this.notesCircle.navItems[0].sliceSelectedAttr.fill = "#c8C8C8";
```

Comment:
at a particular position(i)  you are setting   value of nav items to a particular colour.
You can use a function.  which changes `this.notesCircle.navItems[pos].fillAttr = COLOUR` and call it in the code.
It will be a lot cleaner and maintainable in future.

`setNotesCircleColour(pos,colour)`

Reply:
Yeah, I will work on this.

Reply:
Here, 8 functions would be needed, 4 for `notesCircle` and 4 for `wheel1` each. 
Would it be appropriate?

Reply:
Done.